core_lib/src/overlaypainter.cpp

```diff
@@ -0,0 +1,313 @@
+    painter.setWorldMatrixEnabled(!mOptions.mIsCamera);
+    painter.setBrush(Qt::NoBrush);
+    QPainter::RenderHints previous_renderhints = painter.renderHints();
+    painter.setRenderHint(QPainter::Antialiasing, false);
+
+    painter.drawLine(rect.x(), static_cast<int>(rect.y() + (rect.height() * 0.38)), rect.right(), static_cast<int>(rect.y() + (rect.height() * 0.38)));
+    painter.drawLine(rect.x(), static_cast<int>(rect.y() + (rect.height() * 0.62)), rect.x() + rect.width(), static_cast<int>(rect.y() + (rect.height() * 0.62)));
+    painter.drawLine(static_cast<int>(rect.x() + rect.width() * 0.38), rect.y(), static_cast<int>(rect.x() + rect.width() * 0.38), rect.bottom());
+    painter.drawLine(static_cast<int>(rect.x() + rect.width() * 0.62), rect.y(), static_cast<int>(rect.x() + rect.width() * 0.62), rect.bottom());
+
+    painter.setRenderHints(previous_renderhints);
+    painter.restore();
+}
+
+void OverlayPainter::paintOverlaySafeAreas(QPainter &painter)
+{
+    QRect rect = mOptions.mRect;
+
+    painter.save();
+    painter.setCompositionMode(QPainter::RasterOp_NotSourceAndNotDestination);
+    QPen pen(Qt::DashLine);
+    qreal space = 10;
+    QVector<qreal> dashes;
+    dashes << 10 << space << 10 << space << 10 << space;
+    pen.setDashPattern(dashes);
+    pen.setCosmetic(true);
+    painter.setPen(pen);
+    painter.setWorldMatrixEnabled(!mOptions.mIsCamera);
+    painter.setBrush(Qt::NoBrush);
+    QPainter::RenderHints previous_renderhints = painter.renderHints();
+    painter.setRenderHint(QPainter::TextAntialiasing, true);
+
+    if (mOptions.bActionSafe)
+    {
+        int action = mOptions.nActionSafe;
+        QRect safeAction = QRect(rect.x() + rect.width()*action/200, rect.y() + rect.height()*action/200, rect.width()*(100-action)/100, rect.height()*(100-action)/100);
+        painter.drawRect(safeAction);
+
+        if (mOptions.bShowSafeAreaHelperText) {
+            painter.drawText(safeAction.x(), safeAction.y()-1, tr("Safe Action area %1 %").arg(action));
+        }
+    }
+    if (mOptions.bTitleSafe)
+    {
+        int title = mOptions.nTitleSafe;
+        QRect safeTitle = QRect(rect.x() + rect.width()*title/200, rect.y() + rect.height()*title/200, rect.width()*(100-title)/100, rect.height()*(100-title)/100);
+        painter.drawRect(safeTitle);
+
+        if (mOptions.bShowSafeAreaHelperText) {
+            painter.drawText(safeTitle.x(), safeTitle.y()-1, tr("Safe Title area %1 %").arg(title));
+        }
+    }
+
+    painter.setRenderHints(previous_renderhints);
+    painter.restore();
+}
+
+void OverlayPainter::paintOverlayPerspective1(QPainter &painter)
+{
+    QRect rect = mOptions.mRect;
+
+    painter.save();
+    initPerspectivePainter(painter);
+    QPainter::RenderHints previous_renderhints = painter.renderHints();
+    painter.setRenderHint(QPainter::Antialiasing, false);
+
+    qreal degrees = static_cast<qreal>(mOptions.nOverlayAngle);
+    if (degrees == 7.0) { degrees = 7.5; }
```

Comment:
Any reason the degree cannot be 7.0?

Reply:
In theory - no.
For simplicity and ease of use, I only use angles that are divisors to 180. Right now 2, 3, 5, 7.5, 10, 15, 20 and 30 degrees. We could add 1, 4, 6, 9, 12 and 18.
I we allow the user to pick a random degree, let's say 9.1 degrees, we would have to draw 39.56 angles in 1 point perspective mode. How do we do that?
The same goes for 2 and 3 point perspective modes. When the numbers doesn't add up, it just complicates things.
If we should leave it to the user to decide, we could have a integer spinbox, where the user could enter the number of angles he wanted. If we go for this solution (that I'm no fan of), I think we should have a maximum of 360 angles in 1 point pespective and 180 angles in 2 and 3 point perspective.